tests/gp_tests/test_gp.py

```diff
@@ -6,9 +6,42 @@
 
 from optuna._gp.gp import _fit_kernel_params
 from optuna._gp.gp import KernelParamsTensor
+from optuna._gp.gp import warn_and_convert_inf
 import optuna._gp.prior as prior
 
 
+@pytest.mark.parametrize(
+    "values,ans",
+    [
+        (np.array([-1, 0, 1]), np.array([-1, 0, 1])),
+        (np.array([-1, -np.inf, 0, np.inf, 1]), np.array([-1, -1, 0, 1, 1])),
+        (np.array([[-1, 2], [0, -2], [1, 0]]), np.array([[-1, 2], [0, -2], [1, 0]])),
+        (
+            np.array([[-1, 2], [-np.inf, np.inf], [0, -np.inf], [np.inf, -2], [1, 0]]),
+            np.array([[-1, 2], [-1, 2], [0, -2], [1, -2], [1, 0]]),
+        ),
+        (
+            np.array(
+                [
+                    [-100, np.inf, 10],
+                    [-np.inf, np.inf, 100],
+                    [-10, -np.inf, np.inf],
+                    [np.inf, np.inf, -np.inf],
+                ]
+            ),
+            np.array([[-100, 0, 10], [-100, 0, 100], [-10, 0, 100], [-10, 0, 10]]),
+        ),
+        (np.array([-np.inf, np.inf]), np.array([0, 0])),
+        (np.array([]), np.array([])),
+    ],
+)
+def test_warn_and_convert_inf(values: np.ndarray, ans: np.ndarray) -> None:
+    assert np.allclose(warn_and_convert_inf(values), ans)
+    if len(values.shape) == 1:
+        # Test also with the shape of (n, 1) to ensure the batched version.
+        assert np.allclose(warn_and_convert_inf(values[:, np.newaxis]), ans[:, np.newaxis])
```

Comment:
This test code would be more readable if it were split into two test cases: one for single-dimensional arrays and another for two-dimensional arrays. What do you think?

Reply:
I addressed your comment, PTAL!